tests/ui/base_driver.py

```diff
@@ -1,11 +1,52 @@
 """Base driver module to inherit from."""
 
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
 class BaseDriver(object):
 
   """Base driver that will be called from all pages and elements."""
 
-  # pylint: disable=too-few-public-methods
+  SIDEBAR = (By.TAG_NAME, 'ufo-sidebar')
+  GENERIC_DIV = (By.TAG_NAME, 'div')
 
   def __init__(self, driver):
-    """Create the base driver object."""
+    """Create the base driver object.
+
+    Args:
+      driver: A webdriver instance that can be used to query the dom.
+    """
     self.driver = driver
+
+
+  def GetLink(self, link_locator):
+    """Get an element in the page based on the link locator given.
+
+    Args:
+      link_locator: A tuple of By.SomeProperty and a matching property.
+
+    Returns:
+      A link element if found.
+    """
+    return WebDriverWait(self.driver, 10).until(
+        EC.visibility_of_element_located(((link_locator))))
+
+  def GetElement(self, element_locator):
+    """Get an element in the page based on the element locator given.
+
+    Args:
+      element_locator: A tuple of By.SomeProperty and a matching property.
+
+    Returns:
+      A generic element if found.
+    """
+    return self.driver.find_element(*element_locator)
+
+  def GetSidebar(self):
```

Comment:
Can you explain why the GetSidebar() should be here?  It obviously is a great idea for generic GetElement() and GetLink(), but not quite for something specific like this.

In other words, can we test with something like:
landing_page.GetElement(sidebar.SIDEBAR)?
where SIDEBAR = (By.TAG_NAME, 'ufo-sidebar') is in sidebar.py?